packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -215,11 +220,7 @@ Future<int> startApp(
   }
 
   if (!startedSomething) {
-    if (!devices.all.any((device) => device.isConnected())) {
-      printError('Unable to run application - no connected devices.');
-    } else {
-      printError('Unable to run application.');
-    }
+    printError('Unable to run application. Please ensure that a supported device/simulator is connected.');
```

Comment:
We should retain the different errors messages for when there are no connected devices, and when there was a connected device, but some failure running the application.